frontend/src/TransitPredictionsPage.js

```diff
@@ -79,29 +102,29 @@ export default function TransitPredictionsPage() {
         </div>
         {/* Tabs for larger screens */}
         <div className="hidden md:flex flex-row justify-center space-x-4 overflow-x-auto">
-          {stopOptions.map((stopName) => (
-            <button key={stopName} className={`px-4 py-2 text-xl font-semibold border-b-2 
+          {stopOptions.map((stopName, index) => (
+            <button key={index} className={`px-4 py-2 text-xl font-semibold border-b-2 
             ${selectedStop === stopName ? 'border-blue-500 text-blue-500' : 'border-transparent text-gray-500 hover:border-gray-300'}`}
-            onClick={() => setSelectedStop(stopName)}>
+            onClick={() => changeTab(stopName)}>
               {stopName}
             </button>
           ))}
         </div>
         <div>
-          {!!busData.length && busData.map((stop) => (
+          {!!busData.length && busData.map((stop, index) => (
             stop.name === selectedStop &&
-              <div key={stop.route} className="flex flex-col mt-4 p-6 min-w-80 max-h-[55vh] text-xl overflow-y-auto
+              <div key={index} className="flex flex-col mt-4 p-6 min-w-80 max-h-[55vh] text-xl overflow-y-auto
                 bg-white rounded-lg border border-gray-200 shadow-md dark:bg-gray-800 dark:border-gray-700">
-                <div className="font-bold text-4xl mb-10">{stop.name}</div>
+                <div key={stop.name} className="font-bold text-4xl mb-10">{stop.name}</div>
```

Comment:
i think we only need the key in the parent component?

Reply:
resolved